numpy/linalg/linalg.py

```diff
@@ -2180,11 +2180,14 @@ def lstsq(a, b, rcond="warn"):
     Euclidean 2-norm :math:`||b - ax||`. If there are multiple minimizing 
     solutions, the one with the smallest 2-norm :math:`||x||` is returned.
 
+    .. versionchanged:: 1.19
+       Can now operate on stacks of matrices
+
     Parameters
     ----------
-    a : (M, N) array_like
+    a : (..., M, N) array_like
         "Coefficient" matrix.
-    b : {(M,), (M, K)} array_like
+    b : {(M,), (..., M, K)} array_like
```

Comment:
`numpy.linalg.solve` has the following description for `b` : `{(…, M,), (…, M, K)}, array_like`. `solve` works for the `(…, M,)` case, while `lstsq` from this PR raises `LinAlgError: Incompatible dimensions`.

Should the behavior of this function follow the `solve`?

Reply:
That's probably a good idea, I hadn't realized that we can make that change without affecting compatibility. We'd replace the current `is_1d` logic with something like

https://github.com/numpy/numpy/blob/29561ed6bc96bc7472e43982e13bb96ae3c1a1f5/numpy/linalg/linalg.py#L384-L389